enterprise/api_client/enterprise.py

```diff
@@ -44,12 +47,16 @@ def get_content_metadata(self, enterprise_customer, enterprise_catalogs=None):
         content_metadata = OrderedDict()
         enterprise_customer_catalogs = enterprise_catalogs or enterprise_customer.enterprise_customer_catalogs.all()
         for enterprise_customer_catalog in enterprise_customer_catalogs:
-            response = self._load_data(
-                self.ENTERPRISE_CUSTOMER_CATALOGS_ENDPOINT,
-                resource_id=str(enterprise_customer_catalog.uuid),
-                traverse_pagination=True,
-                querystring={'page_size': 1000},
-            )
+            if switch_is_active('ENTERPRISE_CATALOG_API_ENDPOINTS_ENABLED'):
+                response = \
+                    EnterpriseCatalogApiClient.enterprise_catalog_get_content_metadata(enterprise_customer_catalog.uuid)
```

Comment:
I think we need create `EnterpriseCatalogApiClient` object like `EnterpriseCatalogApiClient(user=self.user)`

Reply:
agreed - ideally the flag would be at the points where we instantiate this client or the other and make the call to it.